src/vaibhav/Assignment_14/MaggiPanipuriByScannerClass.java

```diff
@@ -0,0 +1,88 @@
+package vaibhav.Assignment_14;
+
+import java.util.Scanner;
+
+import vaibhav.Assignment_7.Shop;
+
+public class MaggiPanipuriByScannerClass {
+	
+int iMaggie, iPanipuri;
+	
+	void setInitialQualtiy (int qMaggie, int qPanipuri) {
+		iMaggie = qMaggie;
+		iPanipuri = qPanipuri;
+	}
+		
+	void requestedItems( int reqMaggie , int reqPanipuri, boolean pAllowed) {
+		
+		/// Maggie Check
+		if (iMaggie == 0) {
+			System.out.println("Maggie is running out of stock");		
+		}else if (reqMaggie <= iMaggie) {
+			iMaggie = iMaggie - reqMaggie;
+		}else {
+			System.out.println("Insufficient Maggie available in stock. We can serve you only " + iMaggie + " Maggie out of " + reqMaggie );
+			if (pAllowed)		// pAllowed == true OR pAllowed == false
+				iMaggie = 0;
+		}
+				
+				
+		
+		// Panipuri Check
+		if (iPanipuri == 0) {
+			System.out.println("Panipuri is running out of stock");
+		}else if (reqPanipuri <= iPanipuri) {
+			iPanipuri = iPanipuri - reqPanipuri;
+		}else{
+			System.out.println("Insufficient Panipuri available in stock. We can serve you only " + iPanipuri + " Panipuri out of " + reqPanipuri);
+			if (pAllowed)		// pAllowed == true OR pAllowed == false
+				iPanipuri = 0;
+		}
+		
+	
+		
+	}
+		
+	void outOfStockItems() {
+		System.out.println(" ");
+		System.out.println("Out of stock items are : ");
+		
+		if (iMaggie==0 || iPanipuri == 0) {
+			if (iMaggie == 0)
+				System.out.println("Maggie");
+			if (iPanipuri == 0)
+				System.out.println("Panipuri");
```

Comment:
Please add full statement for OOS

Reply:
@harshvegada 
OOS ==> do you mean out of stock for OOS ?
If Yes ,
I had mentioned line 48 for it for all out of stock elements.

Please let me know if I had missed something.